### PR TITLE
Fill out some TODOs in the code intel/code nav glossary

### PR DESCRIPTION
**Pull request by varungandhi-src**

### CODE REVIEW

File: content/departments/engineering/teams/code-intelligence/glossary.md [hunk 1]
```diff
@@ -67,6 +107,16 @@ Examples:
...
 
+### cross-repository code navigation
+
+[Code navigation](#code-navigation) that works across different repositories. For example, some customers store their code across hundreds or even thousands of repositories. Cross-repository code navigation enables such customers to navigate their code as seamlessly as if it were stored in a single repository.
+
+Adding support for cross-repository navigation for a language requires adding the corresponding logic to the language's [indexer](#indexer). The [Sourcegraph docs on indexers](https://docs.sourcegraph.com/code_navigation/references/indexers) cover which indexers support cross-repository code navigation.
+
+### definition
+
+TODO
```
Comment (scalabilitysolved) [inline content/departments/engineering/teams/code-intelligence/glossary.md:118]: Were you going to fill this in too or is it left as a future TODO?

**Review by scalabilitysolved — APPROVED** [review]
Looks good, some helpful additions 👍 

Left a single comment about a TODO that was added.

File: content/departments/engineering/teams/code-intelligence/glossary.md (hunk 1 above)
Comment (varungandhi-src) [reply content/departments/engineering/teams/code-intelligence/glossary.md:118]: Future TODO.

**Comment by varungandhi-src** [comment]
@peterguy can you confirm if updated description in this PR is satisfactory or do you have any suggestions on things I should clarify?

**Comment by peterguy** [comment]
Wow, @varungandhi-src, you rolled up your sleeves and dug into the whole glossary! Looks great!

**Pull request merged**